File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/AuthorPicture.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Objects;

@Data
@Entity
@Table(name = "author_picture")
public class AuthorPicture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ap_id")
    private Long apId;

    @Column(name = "ap_picture", nullable = false)
    private String apPicture;

    @OneToOne
    @JoinColumn(name = "a_id", nullable = false, unique = true)
    private Author author;

    @Override
    public int hashCode() {
        return Objects.hash(apId, apPicture);
    }

    @Override
    public String toString() {
        return "AuthorPicture{" +
                "apId=" + apId +
                ", apPicture='" + apPicture + '\'' +
                '}';
    }
}

File: Frontend/manga-world/src/Model/AuthorPicture.ts
import Author from "./Author";

class AuthorPicture {
  apId: number;
  apPicture: string;
  author: Author;

  constructor(apId: number, apPicture: string, author: Author) {
    this.apId = apId;
    this.apPicture = apPicture;
    this.author = author;
  }
}

export default AuthorPicture;
